src/backend/parser/parse_agg.c

```diff
@@ -1507,14 +1544,145 @@ get_groupclause_exprs(Node *grpcl, List *targetList)
 
 		foreach (lc, exprs)
 		{
-			result = list_concat(result, get_groupclause_exprs((Node *)lfirst(lc),
+			result = list_concat(result, get_groupclause_tles((Node *)lfirst(lc),
 															   targetList));
 		}
 	}
 
 	return result;
 }
 
+static List*
+get_com_grouping_ressortgroupref_routine(Node *grpcl, List *targetList, List *com_refs)
+{
+
+	List *result = NIL;
+
+	if ( !grpcl)
+		return result;
+
+	Assert(IsA(grpcl, SortGroupClause) ||
+		   IsA(grpcl, GroupingClause) ||
+		   IsA(grpcl, List));
+
+	if (IsA(grpcl, SortGroupClause))
+	{
+		TargetEntry *tle = get_sortgroupclause_tle((SortGroupClause *) grpcl, targetList);
+		result = lappend_int(result, tle->ressortgroupref);
+	}
+
```

Comment:
excessive line

Reply:
I delete similar space in "get_groupclause_tles" too.